sky/clouds/gcp.py

```diff
@@ -512,37 +550,36 @@ def check_credentials(cls) -> Tuple[bool, Optional[str]]:
             ]:
                 if not os.path.isfile(os.path.expanduser(file)):
                     raise FileNotFoundError(file)
+        except FileNotFoundError as e:
+            return False, (
+                f'Credentails are not set. '
+                f'{cls._CREDENTIAL_HINT}\n'
+                f'{cls._INDENT_PREFIX}Details: '
+                f'{common_utils.format_exception(e, use_bracket=True)}')
 
+        try:
             cls._find_application_key_path()
+        except FileNotFoundError as e:
+            return False, (
+                f'Application credentials are not set. '
+                f'{cls._APPLICATION_CREDENTIAL_HINT}\n'
+                f'{cls._INDENT_PREFIX}Details: '
+                f'{common_utils.format_exception(e, use_bracket=True)}')
 
-            # Check the installation of google-cloud-sdk.
-            _run_output('gcloud --version')
-
+        try:
             # Check if application default credentials are set.
             project_id = cls.get_project_id()
 
             # Check if the user is activated.
             identity = cls.get_current_user_identity()
         except (auth.exceptions.DefaultCredentialsError,
-                subprocess.CalledProcessError,
-                exceptions.CloudUserIdentityError, FileNotFoundError,
-                ImportError) as e:
+                exceptions.CloudUserIdentityError) as e:
             # See also: https://stackoverflow.com/a/53307505/1165051
             return False, (
-                'GCP tools are not installed or credentials are not set. '
-                'Run the following commands:\n    '
-                # Install the Google Cloud SDK:
-                '  $ pip install google-api-python-client\n    '
-                '  $ conda install -c conda-forge google-cloud-sdk -y\n    '
-                # This authenticates the CLI to make `gsutil` work:
-                '  $ gcloud init\n    '
-                # This will generate
-                # ~/.config/gcloud/application_default_credentials.json.
-                '  $ gcloud auth application-default login\n    '
-                'For more info: '
-                'https://skypilot.readthedocs.io/en/latest/getting-started/installation.html'  # pylint: disable=line-too-long
-                f'\nDetails: {common_utils.format_exception(e, use_bracket=True)}'
-            )
+                'Credentials are not correctly set. '
```

Comment:
Can we also hint the CLI commands that are equivalent to L571 & 574? Something like "Getting project ID / current user failed. You can debug with "cmd1" and "cmd2". {_CREDENTIAL_HINT}"

Reply:
Good point! Added. There is no exactly equivalent command for `auth.default()`, but this exception is a very rare case, if the previous checks pass. We only add the `gcloud auth list` command here, and can wait for more feedbacks if user encounter this issue. : )